ocaml/typing/typecore.ml

```diff
@@ -3392,6 +3417,11 @@ let collect_apply_args env funct ignore_labels ty_fun ty_fun0 mode_fun sargs ret
             (Eliminated_optional_arg
                { mode_fun; ty_arg; mode_arg; sort_arg; level = lv })
         in
+        let omit_arg () =
+          may_warn funct.exp_loc
+            (Warnings.Non_principal_labels "commuted an argument");
+          Omitted { mode_fun; ty_arg; mode_arg; level = lv; sort_arg }
+        in
```

Comment:
Why factor this out? It seems like it's just used once.

Reply:
It seems that the other two cases are factored out (`eliminate_arg` and `use_arg`), so I factored this out for symmetry.